java/pom.xml

```diff
@@ -628,5 +627,95 @@
+                <maven.compiler.source>1.8</maven.compiler.source>
+                <maven.compiler.target>1.8</maven.compiler.target>
+            </properties>
+
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-javadoc-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <goals>
+                                    <goal>jar</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-source-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>attach-sources</id>
+                                <goals>
+                                    <goal>jar</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-deploy-plugin</artifactId>
+                        <configuration>
+                            <javadoc>true</javadoc>
+                            <sources>true</sources>
+                        </configuration>
+                    </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-release-plugin</artifactId>
+                        <configuration>
+                            <tagNameFormat>java-@{project.version}</tagNameFormat>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+
+            <reporting>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-javadoc-plugin</artifactId>
+                    </plugin>
+                </plugins>
+            </reporting>
+
+            <repositories>
+                <repository>
+                    <id>${releaseRepoId}</id>
+                    <url>${releaseRepoUrl}</url>
+                    <name>${releaseRepoName}</name>
+                </repository>
+            </repositories>
+            <distributionManagement>
+                <repository>
+                    <id>${releaseRepoId}</id>
+                    <url>${releaseRepoUrl}</url>
+                    <name>${releaseRepoName}</name>
+                </repository>
+            </distributionManagement>
+
+        </profile>
     </profiles>
+
+    <scm>
```

Comment:
In addition to <scm/> Microsoft recommends copyright, <developer/> and <license/> elements. See https://docs.opensource.microsoft.com/releasing/publish-binaries/maven/

Aren't these elements usually at the beginning of a pom? Not that order matters. I wonder if it would be worth applying something like tidy-maven-plugin to our poms.